src/core/geometry/qgssfcgalgeometry.h

```diff
@@ -0,0 +1,516 @@
+#include "qgsmultipolygon.h"
+#include "qgsmultisurface.h"
+#include "qgspolyhedralsurface.h"
+#include "qgstriangulatedsurface.h"
+#include "qgstriangle.h"
+#include "qgssfcgalengine.h"
+#include "qgslogger.h"
+
+/**
+ * Wraps SFCGAL geometry object and its QgsAbstractGeometry source
+ * \ingroup core
+ * \class QgsSfcgalGeometry
+ * \brief SfcgalGeometry geometry type.
+ * \since QGIS 3.44
+ */
+class CORE_EXPORT QgsSfcgalGeometry
+{
+  public:
+    //! Constructor for an empty sfcgalGeometry geometry.
+    QgsSfcgalGeometry();
+
+    //! Constructor with SFCGAL shared ptr
+    QgsSfcgalGeometry( sfcgal::shared_geom sfcgalGeom );
+
+    //! Constructor with QgsAbstractGeometry unique_ptr
+    QgsSfcgalGeometry( std::unique_ptr<QgsAbstractGeometry> &qgsGeom );
+
+    //! Constructor with QgsAbstractGeometry ptr
+    QgsSfcgalGeometry( const QgsAbstractGeometry &qgsGeom );
+
+    //! Constructor with QgsAbstractGeometry ptr
+    QgsSfcgalGeometry( const QgsGeometry &qgsGeom );
+
+    //! Constructor from WKT
+    QgsSfcgalGeometry( const QString &wkt, QString *errorMsg = nullptr );
+
+    //! Copy constructor
+    QgsSfcgalGeometry( const QgsSfcgalGeometry &otherGeom );
+
+    //! Returns the underlying SFCGAL geometry
+    sfcgal::shared_geom sfcgalGeometry() const { return mSfcgalGeom; }
+
+    //! Returns the underlying QGIS geometry
+    QgsAbstractGeometry *asQgisGeometry( QString *errorMsg = nullptr ) const;
+
+    /**
+     * Returns type of the geometry as a WKB type (point / linestring / polygon etc.)
+     * \param errorMsg Error message returned by SFGCAL
+     */
+    Qgis::WkbType wkbType( QString *errorMsg = nullptr ) const;
+
+    /**
+     * Returns type of the geometry as String
+     * \param errorMsg Error message returned by SFGCAL
+     */
+    QString geometryType( QString *errorMsg = nullptr ) const SIP_HOLDGIL;
+
+    /**
+     * Clones the geometry by performing a deep copy
+     */
+    QgsSfcgalGeometry *clone() const;
+
+    /**
+     * Creates a new geometry from a WKB byte pointer
+     * \param wkbPtr WKB byte pointer
+     * \param errorMsg Error message returned by SFGCAL
+     */
+    bool fromWkb( QgsConstWkbPtr &wkbPtr, QString *errorMsg = nullptr );
+
+    /**
+     * Export the geometry as WKB
+     *
+     * \param flags argument specifies flags controlling WKB export behavior (since QGIS 3.14).
+     * \param errorMsg Error message returned by SFGCAL
+     */
+    QByteArray asWkb( QgsAbstractGeometry::WkbFlags flags = QgsAbstractGeometry::WkbFlags(), QString *errorMsg = nullptr ) const;
+
+    /**
+     * Creates a new geometry from a WKT string.
+     * \param wkt WTK string
+     * \param errorMsg Error message returned by SFGCAL
+     */
+    bool fromWkt( const QString &wkt, QString *errorMsg = nullptr );
+
+    /**
+     * Export the geometry as WKT
+     *
+     * \param precision Floating point precision for WKT coordinates. Setting to -1 yields rational number WKT (not decimal).
```

Comment:
what does "rational name" mean here? Does it return fraction text (Eg "Point ( 1/3 , 2/3)" ?)

Reply:
That's not actually WKT then, is it? At least the QGIS wkt parser won't accept fraction strings...

Reply:
Yes I can set the default precision to any positive number to obtain, by default, something compatible with QGIS?